applications/main/nfc/scenes/nfc_scene_mf_classic_detect_reader.c

```diff
@@ -0,0 +1,150 @@
+#include "../nfc_app_i.h"
+
+#include <nfc/protocols/mf_classic/mf_classic_listener.h>
+
+#define NFC_SCENE_DETECT_READER_PAIR_NONCES_MAX (10U)
+#define NFC_SCENE_DETECT_READER_WAIT_NONCES_TIMEOUT_MS (1000)
+
+static const NotificationSequence sequence_detect_reader = {
+    &message_green_255,
+    &message_blue_255,
+    &message_do_not_reset,
+    NULL,
+};
+
+void nfc_scene_mf_classic_detect_reader_view_callback(void* context) {
+    NfcApp* instance = context;
+
+    view_dispatcher_send_custom_event(instance->view_dispatcher, NfcCustomEventViewExit);
+}
+
+NfcCommand nfc_scene_mf_classic_detect_listener_callback(NfcGenericEvent event, void* context) {
+    furi_assert(context);
+    furi_assert(event.event_data);
+    furi_assert(event.protocol == NfcProtocolMfClassic);
+
+    NfcApp* instance = context;
+    MfClassicListenerEvent* mfc_event = event.event_data;
+
+    if(mfc_event->type == MfClassicListenerEventTypeAuthContextPartCollected) {
+        MfClassicAuthContext* auth_ctx = &mfc_event->data->auth_context;
+        mfkey32_logger_add_nonce(instance->mfkey32_logger, auth_ctx);
+        view_dispatcher_send_custom_event(instance->view_dispatcher, NfcCustomEventWorkerUpdate);
+    }
+
+    return NfcCommandContinue;
+}
+
+void nfc_scene_mf_classic_timer_callback(void* context) {
+    NfcApp* instance = context;
+
+    view_dispatcher_send_custom_event(instance->view_dispatcher, NfcCustomEventTimerExpired);
+}
+
+void nfc_scene_mf_classic_detect_reader_on_enter(void* context) {
+    NfcApp* instance = context;
+
+    if(nfc_device_get_protocol(instance->nfc_device) == NfcProtocolInvalid) {
+        const Iso14443_3aData iso3_data = {
+            .uid_len = 7,
+            .uid = {0x04, 0x77, 0x70, 0x2A, 0x23, 0x4F, 0x80},
```

Comment:
Could we make it really random. I trust the time will come and readers will start filtering by uid